src/metpy/calc/thermo.py

```diff
@@ -3885,20 +3894,83 @@ def dewpoint_from_specific_humidity(pressure, temperature, specific_humidity):
     --------
     >>> from metpy.calc import dewpoint_from_specific_humidity
     >>> from metpy.units import units
-    >>> dewpoint_from_specific_humidity(1000 * units.hPa, 10 * units.degC, 5 * units('g/kg'))
-    <Quantity(3.73203192, 'degree_Celsius')>
+    >>> dewpoint_from_specific_humidity(1000 * units.hPa, 5 * units('g/kg'))
+    <Quantity(3.79314079, 'degree_Celsius')>
+
+    .. versionchanged:: 1.6
+       Made `temperature` arg optional, to be deprecated
 
     .. versionchanged:: 1.0
        Changed signature from ``(specific_humidity, temperature, pressure)``
 
     See Also
     --------
-    relative_humidity_from_mixing_ratio, dewpoint_from_relative_humidity
+    relative_humidity_from_mixing_ratio, dewpoint_from_relative_humidity, dewpoint
+
+    Notes
+    -----
+    Employs [WMO8]_ eq 4.A.6,
+
+    .. math:: e = \frac{w}{\epsilon + w} p
+
+    with
+
+    .. math:: w = \frac{q}{1-q}
+
+    * :math:`q` is specific humidity
+    * :math:`w` is mixing ratio
+    * :math:`\epsilon` is the molecular weight ratio of vapor to dry air
 
+    to calculate vapor partial pressure :math:`e` for dewpoint calculation input. See
+    :func:`~dewpoint` for additional information.
     """
-    return dewpoint_from_relative_humidity(temperature,
-                                           relative_humidity_from_specific_humidity(
-                                               pressure, temperature, specific_humidity))
+    if (len(args) + len(kwargs)) > 2:
+        _warnings.warn(
+            'Temperature argument is unused and will be deprecated in a future version.',
+            PendingDeprecationWarning)
+
+    pressure = kwargs.pop('pressure', None)
+    specific_humidity = kwargs.pop('specific_humidity', None)
+
+    return _dewpoint_from_specific_humidity(
+        pressure if pressure else args[0],
+        specific_humidity if specific_humidity else args[-1])
```

Comment:
Right now this is going to ignore any argument passed in as a keyword argument. I'm 99% sure that doing:
```python
dewpoint_from_specific_humidity(pressure, specific_humidity=q)
```
will result in `pressure` being passed for both arguments to `_dewpoint_from_specific_humidity()`. Instead we need to do:
```
if len(args) > 1 or 'temperature' in kwargs:
   kwargs.pop('temperature')
   args = args[:1]
   ...
```
then
```python
import inspect
sig = inspect.Signature(_dewpoint_from_specific_humidity)
args = sig.bind(*args, **kwargs)
return _dewpoint_from_specific_humidity(pressure=args.pressure, args.specific_humidity)
```
That's untest, but I think will work.

Reply:
This PR includes a [test](https://github.com/Unidata/MetPy/blob/a418f4d6c20da60b68a0daec7e8f5c522ed524a1/tests/calc/test_thermo.py#L1832-L1840) that covers a similar case already, and your snippet does currently work correctly. I'm glad to add your snippet as an additional test. Am I missing a potential hole here?

Reply:
That one has all 3 arguments. So I guess good to add a test that covers only passing two arguments (`pressure` and `specific_humidity`)? Either that test fails or I fail at understanding Python function argument handling.

Reply:
Done!

Reply:
>  Either that test fails or I fail at understanding Python function argument handling.

Option (c), @dopplershift should be less distracted when trying to read and understand code.

Reply:
So I missed that you handled explicitly the `kwargs`.

With that said, there is a real problem: this function chokes on arrays because `if arr` is no bueno for arrays. Either needs to be something like `if arr is not None`. Probably should add a test (or modify one) to include array support.

I'm wondering if using `inspect.Signature` and `bind()` might be the simpler overall approach?

Reply:
Ah, good catch. I'll go add a test and take a look at you suggested implementation

Reply:
Double check my example code, I got it wrong the first time.

Reply:
Thanks for the suggested update to the handling. Should be good to go now, and added some new tests!